clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1902,12 +1904,12 @@ protected void recordProcessedRecordStats(
 
   @Override
   protected final double calculateAssembledRecordSizeRatio(long recordSize) {
-    return (double) recordSize / maxRecordSizeBytes;
+    return (double) recordSize / maxNearlineRecordSizeBytes;
```

Comment:
It seems that we don't differentiate batch record from the nearline data in calculating the ratio. Would that be okay? What's your opinion?

Reply:
That's true. We already cannot differentiate between batch and nearline data when calculating the metric. Now there is an added difference in the record size limit between batch and nearline. The default batch limit goes through the controller path, so the server (this code path) never even gets to see it.

Maybe we can just leave that as a limitation, especially given the current low number of stores with large records. What do you think @gaojieliu?

Reply:
I think it will be fine to use `maxRecordSizeBytes` (batch part) for metrics purpose, and the reason to introduce `maxNearlineRecordSizeBytes` is to avoid failing the nearline job before we have a good way to communicate with the customers.

Reply:
In theory, we should use the same number for both the batch job and nearline job eventually when everything is ready, and the separate nearline limit is intermediate to ease the feature rollout.

Reply:
Sounds good, please add some comments in the to clarify it.